Allow disabling of storage pods

piyush-nimbalkar opened the pull request:
- If portworx is disabled, then do not deploy
  portworx pods, portworx RBAC, API pods, portworx service, storage classes
- Do not update storage cluster status or create storage node objects
- Do not allow deleting the portworx with any delete strategy
- We should still be able to deploy components using the StorageCluster

Signed-off-by: Piyush Nimbalkar <piyush@portworx.com>

adityadani commented on pkg/controller/storagecluster/storagecluster.go:
```diff
@@ -64,7 +65,10 @@ import (
+	ControllerName = "storagecluster-controller"
+	// AnnotationDisableStorage annotation to disable the storage pods from running.
+	// Defaults to false value.
+	AnnotationDisableStorage            = operatorPrefix + "/disable-storage-pods"
```
nit: why not just "disable-storage "?

piyush-nimbalkar commented:
Added another commit to remove portworx api, service and rbac when portworx is disabled post installation. There is no need to keep the above components as portworx pods are not running anyway. 
Also addressed the comment.
